pkg/monitors/elasticsearch/query/response.go

```diff
@@ -0,0 +1,186 @@
+package query
+
+import (
+	"encoding/json"
+)
+
+type HTTPResponse struct {
+	Aggregations map[string]*aggregationResponse `json:"aggregations"`
+}
+
+// Represents all kinds of supported aggregations. aggregationResponse type can
+// be determined in the following manner -
+// 1) single bucket aggregationInfo contains only "doc_count" and a map to sub aggregations
+// 2) multi bucket aggregationInfo contains "buckets"and a map to sub aggregations
+// 3) metric aggregationInfo contains "value" or "values"
+// These structs are defined based on inputs from
+// https://github.com/elastic/elasticsearch/tree/master/server/src/main/java/org/elasticsearch/search/aggregations
+type aggregationResponse struct {
+	// Number of documents backing the aggregationInfo
+	DocCount *int64 `json:"doc_count,omitempty"`
+	// Non nil for single value metric aggregations
+	Value interface{} `json:"value,omitempty"`
+	// Non nil for multi-value metrics aggregations
+	Values interface{} `json:"values,omitempty"`
+	// Map of sub aggregations with aggregationInfo names, includes both bucket
+	// and metric aggregations
+	SubAggregations map[string]*aggregationResponse `json:"-"`
```

Comment:
Incorporated your suggestion about having types for buckets and aggregations. For the purposes in this monitor I don't know if it would have made a difference but I think in general mapping to references allow for a easier way to modify entries in a map.